### PR TITLE
[#19917] feat: rename keypair from wallet settings

### LINKED ISSUE #19917: Allow users to rename key-pair from wallet settings
# Feature Issue

### User Story

As a user,
I want edit the name of a key-pair while inside the wallet settings.

### Description 
Implement the edit feature for key-pair names inside wallet settings.

### Designs

Figma flow: [the Figma designs](https://www.figma.com/file/hLFPHYnuM0pOzgiabtqQ6k/Descoped-Wallet-settings-for-Mobile?type=design&node-id=49-135407&mode=design&t=1clXftQkm0Uw1oCE-4)

### Notes
* Based on [the designs here](https://www.figma.com/file/hLFPHYnuM0pOzgiabtqQ6k/Descoped-Wallet-settings-for-Mobile?type=design&node-id=49-133839&mode=design&t=1clXftQkm0Uw1oCE-4), the user should not be allowed to rename the default key-pair.

### Acceptance Criteria

| Screen | Description | Acceptance Criteria |
|--------|--------|--------|
| Key-pair Actions Menu | Users can choose to rename a key-pair with actions menu | A user should be able to navigate to the name edit screens for a key-pair. |
| Key-pair Name Edit | Users can rename a key-pair with the edit screen | A user should be able to edit the name of a key-pair. And the key-pair name edit form should only allow saving if the name has been changed |
| Key-pair Name Exists | Users cannot rename a key-pair with an existing key-pair name | The key-pair name edit form should prevent the user from renaming a key-pair with an existing name |
| Key-pair Name Too Long | Users cannot rename a key-pair with too long of a name | The key-pair name edit form should prevent the user from renaming a key-pair with too long of a name |
| Key-pair Name Special Characters | Users cannot rename a key-pair with special characters | The key-pair name edit form should prevent the user from renaming a key-pair with special characters |
| Key-pair Name Emojis | Users cannot rename a key-pair with emojis | The key-pair name edit form should prevent the user from renaming a key-pair with emojis |
| Key-pair Name Valid | Users can rename a key-pair with a valid name | The key-pair name edit form should allow the user to rename a key-pair with a valid name |
| Key-pair Renamed Toast | A toast will appear to confirm a key-pair has been renamed | The user should see a toast that confirms a key-pair has been renamed |
| Key-pair Renamed | Users can see the renamed key-pair in the key-pair and accounts list | The user should see the confirmation toast disappear and should see the renamed key-pair |

### PR DESCRIPTION
**Pull request by mohsen-ghafouri**
fixes #19917

### Summary

Implement the edit feature for key-pair names inside wallet settings.

#### Areas that maybe impacted
- Rename keypair

### Steps to test

- Open Wallet setting
- Click on not default keypair's action button
- Click on rename 

### Result

https://github.com/status-im/status-mobile/assets/71308738/7ba90fc5-f314-46b6-aba0-c44933131019

status: ready

### CODE REVIEW

File: src/status_im/common/validation/keypair.cljs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+(ns status-im.common.validation.keypair
+  (:require [clojure.string :as string]
+            [status-im.common.validation.general :as validators]
+            [status-im.constants :as constants]
+            utils.emojilib
+            [utils.i18n :as i18n]))
+
+(defn keypair-too-short? [s] (< (count (string/trim (str s))) constants/keypair-name-min-length))
+(defn keypair-too-long? [s] (> (count (string/trim (str s))) constants/keypair-name-max-length))
```
Comment (ibrkhalil) [inline src/status_im/common/validation/keypair.cljs]: nit: `->` macro here would make this really readable.

File: src/status_im/common/validation/keypair_test.cljs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+(deftest validation-keypair-name-test
+  (are [arg expected]
+   (= (keypair-validator/validation-keypair-name arg) expected)
+   nil                         nil
+   ""                          nil
+   "name !"                    (i18n/label :t/key-name-error-special-char)
+   "Hello 😊"                  (i18n/label :t/key-name-error-emoji)
+   "abc"                       (i18n/label :t/your-key-pair-name-is-too-short)
+   (apply str (repeat 25 "a")) (i18n/label :t/your-key-pair-name-is-too-long)))
```
Comment (ibrkhalil) [inline src/status_im/common/validation/keypair_test.cljs:27]: noice-meme

File: src/status_im/contexts/settings/wallet/keypairs_and_accounts/rename/view.cljs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            [react-native.core :as rn]
+            [status-im.common.floating-button-page.view :as floating-button-page]
+            [status-im.common.validation.keypair :as keypair-validator]
+            [status-im.constants :as constants]
+            [status-im.contexts.settings.wallet.keypairs-and-accounts.rename.style :as style]
+            [utils.debounce :as debounce]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn navigation-back [] (rf/dispatch [:navigate-back]))
```
Comment (ibrkhalil) [inline src/status_im/contexts/settings/wallet/keypairs_and_accounts/rename/view.cljs]: nit: I'd rename the function name to `navigate-back`

**Review by ibrkhalil — COMMENTED** [review]
✅

**Review by ibrkhalil — APPROVED** [review]

**Review by seanstrom — APPROVED** [review]
Nice stuff ✅

**Comment by mariia-skrypnyk** [comment]
Hi @mohsen-ghafouri !

I am wondering should this PR skip manual testing as you wrote steps to test.
Just want to be sure any mistake here.
Thanks!

I've checked failed e2e and they are not related to your PR.

**Comment by mohsen-ghafouri** [comment]
Hi @mariia-skrypnyk this screen is behind the Feature flag and i think we agreed to ask for manual test on feature flag removal pull request. but if you want to do manual test now it's ready as well. 

CC @churik

**Comment by mariia-skrypnyk** [comment]
Thanks @mohsen-ghafouri!
You are wright.
Great you put steps for future testing purposes.

You can merge 🚀!

**Pull request merged**